### PR TITLE
Replace flake8 with black

### PR DESCRIPTION
**Pull request by millatidy**
Replace flake8 with black

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,9 @@ clean-pyc: ## Remove Python file artifacts
 clean: clean-tox clean-build clean-pyc ## Remove all file artifacts
 
 .PHONY: lint
-lint: ## Check code style with flake8
+lint: ## Check code style with black
 	@echo "+ $@"
-	@tox -e flake8
+	@tox -e black
```
Comment (michaeljoseph) [inline Makefile]: I don't _think_ we have a toxenv called `black`. `lint` maybe?

**Comment by michaeljoseph** [comment]
Not entirely sure I agree with this PR, `flake8` can do a lot more things than just paint it black?

File: test_requirements.txt [hunk 2]
```diff
@@ -3,3 +3,4 @@ pytest-cov
 pytest-mock
 pytest-catchlog
 freezegun
+black
```
Comment (michaeljoseph) [inline test_requirements.txt]: We probably don't need this because `pre-commit` will install `black`?

**Comment by hackebrot** [comment]
Hi @millatidy! 👋 

Thank you for your pull-request to Cookiecutter! 🍪

I'm afraid I'm with @michaeljoseph here. I'm happy to auto-format source code with black, but we will need to continue running flake8 as a code linter.

**Comment by millatidy** [comment]
> Hi @millatidy! 👋
> 
> Thank you for your pull-request to Cookiecutter! 🍪
> 
> I'm afraid I'm with @michaeljoseph here. I'm happy to auto-format source code with black, but we will need to continue running flake8 as a code linter.

I noticed some PR's are failing due to vague linting issues. That is why I made this PR. How the the failing PR's be fixed for instance this PR?

**Comment by insspb** [comment]
I played with black a little bit and have some thoughts about it. 
1. It is not style checker, it is formatter. So we should not change flake8 for it. Moreover in next versions of python we, I hope, will extend flake8 with mypy type checking plugin.
2. Implementation of black with default settings will require a complete codebase code change. That will make all existing pull requests problematic. Here is example of what black with do with one of our files, when black allowed to any change.
Screenshot from 2019-08-25 17-11-09
3. In other places, black sometimes do a good job. (With restricted quotes change) I like this type of formatting:
Screenshot from 2019-08-25 17-10-02
But if line not so long it can break already formatted to oneliner. I do not like it for lines where it was already formatted vertically.
Screenshot from 2019-08-25 17-10-24

So at the end of the day. this commit will break existing codebase and will require next commit to fix it. 
This commit is not ready to merge.
Black CAN be implemented as precommit hook on github, and only in this way. But this should be done when most current PRs are closed/merged.
I will close it. You can open an issue to discuss this feature.

**Pull request closed without merging**